### ISSUE TITLE
zombie VMs on dair-qc generating tracebacks

### ISSUE THREAD

**Issue by rseuster** [hep-gc/cloudscheduler#117]
there seems to be communication problems with dair-qc. from the openstack dashboard, there are no VMs running anymore.
csv2 still thinks they are running:
```
VMs: (2/3)
+---------+---------+-------------------------------------------+-----------+------------+--------------+-------------+----------------+---------------------+---------------------+----------+
+ Group   | Cloud   | Hostname                                  | Flavor ID | Task       | Power Status | Terminate   | Terminate Time | Status Change Time  | Last Updated        | Flavor   +
+---------+---------+-------------------------------------------+-----------+------------+--------------+-------------+----------------+---------------------+---------------------+----------+
| testing | dair-qc | testing--dair-qc--1685--204336270835202   | 10        | scheduling | 0            | 1           | None           | 2018-12-04 22:01:12 | 2018-12-04 19:54:12 | m1.small |
| testing | dair-qc | testing--dair-qc--1685--51781721313584-1  | 9         | scheduling | 0            | 1           | None           | 2018-12-04 22:01:12 | 2018-12-04 17:08:47 | None     |
| testing | dair-qc | testing--dair-qc--1685--76986040180356    | 10        | scheduling | 0            | 1           | None           | 2018-12-04 22:01:12 | 2018-12-04 19:55:49 | m1.small |
| testing | dair-qc | testing--dair-qc--1685--97121175818833-1  | 9         | scheduling | 0            | 1           | None           | 2018-12-04 22:01:12 | 2018-12-04 19:32:57 | None     |
| testing | dair-qc | testing--dair-qc--1685--170684841653087   | 9         | scheduling | 0            | 1           | None           | 2018-12-04 22:01:12 | 2018-12-04 19:34:07 | None     |
| testing | dair-qc | testing--dair-qc--1685--13626952882196-1  | 9         | scheduling | 0            | 1           | None           | 2018-12-04 22:01:12 | 2018-12-04 16:52:01 | None     |
| testing | dair-qc | testing--dair-qc--1685--97121175818833-2  | 9         | scheduling | 0            | 1           | None           | 2018-12-04 22:01:12 | 2018-12-04 19:32:57 | None     |
| testing | dair-qc | testing--dair-qc--1685--276018393714098-1 | 9         | scheduling | 0            | 1           | None           | 2018-12-04 22:01:12 | 2018-12-04 19:33:31 | None     |
| testing | dair-qc | testing--dair-qc--1685--13626952882196-4  | 9         | scheduling | 0            | 1           | None           | 2018-12-04 22:01:12 | 2018-12-04 16:52:01 | None     |
| testing | dair-qc | testing--dair-qc--1685--12268771189755    | 10        | scheduling | 0            | 1           | None           | 2018-12-04 22:01:12 | 2018-12-04 19:53:13 | m1.small |
| testing | dair-qc | testing--dair-qc--1685--69836985279296-1  | 9         | scheduling | 0            | 1           | None           | 2018-12-04 22:01:12 | 2018-12-04 16:51:24 | None     |
| testing | dair-qc | testing--dair-qc--1685--13626952882196-2  | 9         | scheduling | 0            | 1           | None           | 2018-12-04 22:01:12 | 2018-12-04 16:52:01 | None     |
| testing | dair-qc | testing--dair-qc--1685--69836985279296-4  | 9         | scheduling | 0            | 1           | None           | 2018-12-04 22:01:12 | 2018-12-04 16:51:24 | None     |
| testing | dair-qc | testing--dair-qc--1685--126664016938950   | 10        | scheduling | 0            | 1           | None           | 2018-12-04 22:01:12 | 2018-12-04 19:53:42 | m1.small |
| testing | dair-qc | testing--dair-qc--1685--69836985279296-2  | 9         | scheduling | 0            | 1           | None           | 2018-12-04 22:01:12 | 2018-12-04 16:51:24 | None     |
| testing | dair-qc | testing--dair-qc--1685--51781721313584-2  | 9         | scheduling | 0            | 1           | None           | 2018-12-04 22:01:12 | 2018-12-04 17:08:47 | None     |
| testing | dair-qc | testing--dair-qc--1685--69836985279296-3  | 9         | scheduling | 0            | 1           | None           | 2018-12-04 22:01:12 | 2018-12-04 16:51:24 | None     |
| testing | dair-qc | testing--dair-qc--1685--63452664472835    | 10        | scheduling | 0            | 1           | None           | 2018-12-04 22:01:12 | 2018-12-04 19:55:12 | m1.small |
| testing | dair-qc | testing--dair-qc--1685--51781721313584-4  | 9         | scheduling | 0            | 1           | None           | 2018-12-04 22:01:12 | 2018-12-04 17:08:47 | None     |
| testing | dair-qc | testing--dair-qc--1685--51781721313584-3  | 9         | scheduling | 0            | 1           | None           | 2018-12-04 22:01:12 | 2018-12-04 17:08:47 | None     |
| testing | dair-qc | testing--dair-qc--1685--33623800302231-1  | 10        | scheduling | 0            | 1           | None           | 2018-12-04 22:01:12 | 2018-12-04 19:52:12 | m1.small |
| testing | dair-qc | testing--dair-qc--1685--276018393714098-2 | 9         | scheduling | 0            | 1           | None           | 2018-12-04 22:01:12 | 2018-12-04 19:33:31 | None     |
| testing | dair-qc | testing--dair-qc--1685--81334930915187    | 10        | scheduling | 0            | 1           | None           | 2018-12-04 22:01:12 | 2018-12-04 19:52:42 | m1.small |
| testing | dair-qc | testing--dair-qc--1685--89300402714069    | 10        | scheduling | 0            | 1           | None           | 2018-12-04 22:01:12 | 2018-12-04 19:54:42 | m1.small |
| testing | dair-qc | testing--dair-qc--1685--13626952882196-3  | 9         | scheduling | 0            | 1           | None           | 2018-12-04 22:01:12 | 2018-12-04 16:52:01 | None     |
| testing | dair-qc | testing--dair-qc--1685--33623800302231-2  | 10        | scheduling | 0            | 1           | None           | 2018-12-04 22:01:12 | 2018-12-04 19:52:12 | m1.small |
+---------+---------+-------------------------------------------+-----------+------------+--------------+-------------+----------------+---------------------+---------------------+----------+
```

Now, lots of python tracebacks are in the logs:
```
2018-12-04 21:59:36,076 - VM Commands      - ERROR - Failed to terminate VM: testing--dair-qc--1685--13626952882196-1
Traceback (most recent call last):
  File "/opt/cloudscheduler/data_collectors/openstack/openstackPoller.py", line 168, in command_poller
    nova.servers.delete(vm.vmid)
  File "/usr/lib/python3.6/site-packages/novaclient/v2/servers.py", line 1464, in delete
    return self._delete("/servers/%s" % base.getid(server))
  File "/usr/lib/python3.6/site-packages/novaclient/base.py", line 370, in _delete
    resp, body = self.api.client.delete(url)
  File "/usr/lib/python3.6/site-packages/keystoneauth1/adapter.py", line 319, in delete
    return self.request(url, 'DELETE', **kwargs)
  File "/usr/lib/python3.6/site-packages/novaclient/client.py", line 83, in request
    raise exceptions.from_response(resp, body, url, method)
novaclient.exceptions.NotFound: Instance 2a6fbba2-ccfb-471c-be29-8cfbe1023fea could not be found. (HTTP 404) (Request-ID: req-8b242f86-08c9-4be1-853c-51eba18dc23a)
2018-12-04 21:59:36,076 - VM Commands      - ERROR - Instance 2a6fbba2-ccfb-471c-be29-8cfbe1023fea could not be found. (HTTP 404) (Request-ID: req-8b242f86-08c9-4be1-853c-51eba18dc23a)
```
The code should handle this gracefully.

**Comment by colsond** [comment]
This should be fixed with commit 30840ccbd8ca10335da6

**Comment by rseuster** [comment]
closing
